### PR TITLE
Multiple fixes in digital_ocean_tag module

### LINKED ISSUE #24265: digital_ocean_tag reporting that tag on droplet changed, when it didn't.
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

digital_ocean_tag

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /Users/brad/Projects/systems/vpn/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.10 (default, Feb  6 2017, 23:53:20) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 (clang-800.0.34)]
```

##### OS / ENVIRONMENT

MacOS

##### SUMMARY

When using digital_ocean_tag to create a tag, the task is marked "changed" even if the tag already exists.  When using digital_ocean_tag to apply a tag to a droplet, the task is marked "changed" even if the droplet already has that tag.  

##### STEPS TO REPRODUCE

Simply run the playbook containing that task twice, and notice that it is "changed" the second time.

```yaml
    - name: Tag the droplet
      digital_ocean_tag:
        name: "{{ do_tag }}"
        resource_id: "{{ do.droplet.id }}"
        api_token: "{{ do_token }}"
        state: present

```

##### EXPECTED RESULTS

When the state is "present", and the tag is already present on the droplet, the task result should be "ok", not "changed".

##### ACTUAL RESULTS

The task result is invariably "changed" (even when the droplet tag state did not change).

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
* Pep8 fixes
* Removed redundant check for name
* Check validity of api_token
* Don't report changed when tag is already present

Fixes #24265

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/digital_ocean/digital_ocean_tag.py

##### ANSIBLE VERSION
```
2.4 devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
ready_for_review

**Comment by Akasurde** [comment]
@kontrafiktion Could you please take a look into this ?

**Comment by Akasurde** [comment]
ready_for_review

**Comment by Akasurde** [comment]
@resmo Can you please take a look into this PR ? Thanks.

File: lib/ansible/modules/cloud/digital_ocean/digital_ocean_tag.py [hunk 1]
```diff
@@ -189,45 +189,65 @@ def core(module):
...
+                    url = "tags/{}/resources".format(name)
+                    payload = {
+                        'resources': [{
+                            'resource_id': resource_id,
+                            'resource_type': resource_type}]}
+                    response = rest.post(url, data=payload)
+                    if response.status_code == 204:
+                        module.exit_json(changed=True)
+                    else:
+                        module.fail_json(msg="error tagging resource '{}': {}".format(resource_id, response.json["message"]))
```
Comment (resmo) [inline lib/ansible/modules/cloud/digital_ocean/digital_ocean_tag.py]: IMHO this is not python 2.6 compatible (https://docs.python.org/2/library/string.html#format-string-syntax). I usually avoid format at all and use the % syntax.

**Review by resmo — CHANGES_REQUESTED** [review]
would recommend to replace format syntax with %, but at least make it 2.6 compatible

**Comment by Akasurde** [comment]
@resmo Done.

**Review by resmo — APPROVED** [review]

**Comment by resmo** [comment]
perfect. (realized that format {} was even there before your works...) 
code LGTM, but could not test.

shipit

@kontrafiktion @brad2014 were you able to verify?

**Review by BondAnthony — APPROVED** [review]

**Comment by BondAnthony** [comment]
shipit

**Comment by Akasurde** [comment]
@resmo @BondAnthony @gundalow Thanks for review and merge.

**Comment by BondAnthony** [comment]
Thank you for the contribution @Akasurde. We have more DigitalOcean modules that need to be updated and additional DigitalOcean api functionatilty that needs to included in Ansible.

**Comment by Akasurde** [comment]
@BondAnthony Thanks. Let me know if you need any help from my side.

**Pull request merged**